src/index.js

```diff
@@ -53,9 +53,43 @@ function isReactClass(superClass, scope, globalOptions) {
   return answer
 }
 
+function areSetsEqual(set1, set2) {
+  if (set1 === set2) {
+    return true
+  }
+
+  if (set1.size !== set2.size) {
+    return false
+  }
+
+  return !Array.from(set1).some(item => !set2.has(item))
+}
+
 export default function(api) {
   const { template, types, traverse } = api
 
+  const nestedIdentifiers = new Set()
+  const removedPaths = new Set()
```

Comment:
this contains objects, and is never traversed, so it probably should be a WeakSet